ibis/backends/conftest.py

```diff
@@ -730,3 +731,21 @@ def test_employee_data_2():
     )
 
     return df2
+
+
+@pytest.fixture
+@pytest.mark.usefixtures("backend")
+def no_duckdb(monkeypatch):
+    _duckdb = sys.modules.get("duckdb", None)
+    monkeypatch.setitem(sys.modules, "duckdb", None)
+    yield
+    monkeypatch.setitem(sys.modules, "duckdb", _duckdb)
```

Comment:
The fact that codecov complains that this line is not covered is fishy. Is a try / finally protection required here to avoid introducing tests with a side effect? OR does the `monkeypatch` fixture offers this protection by default?

I seems that since not other duckdb-dependant tests fail, the latter might hold, but still it's confusing.

Reply:
The automatic `finally` behavior is part of `pytest` fixtures.

Reply:
But they why would codecov complain?

Reply:
Unclear, maybe codecov is broken 🤷🏻

Reply:
Honestly I doubt it but it's probably not that important as long as it does not have a side effect break other unrelated tests.

Reply:
Actually, this test is never run when passing any pytest markers:

```
```


I think what's happening is that `usefixtures` doesn't actually pass `backend` as an argument to the fixture (not sure why, we probably are misunderstanding what `usefixtures` does) so this test never gets picked up by our backend test collection code.

Passing `backend` explicitly to `no_duckdb` fixes the issue.

Thanks for keeping us honest @ogrisel!

Reply:
Ah, yeah we're just using it wrong here, it has no effect when applied to a fixture: https://docs.pytest.org/en/latest/reference/reference.html#pytest-mark-usefixtures

Reply:
Pushed up a commit to fix.

Reply:
You also don't need to yield and cleanup after, monkeypatch does all that for you. (Apologies for the brief comment, I'm on my phone).

Reply:
I added the `yield` initially because every backend test suite was failing, glad we sorted out the actual issue.  Time to read more docs.